src/main/java/org/folio/services/budget/CreateBudgetService.java

```diff
@@ -50,18 +55,24 @@ public CreateBudgetService(RestClient restClient,
   }
 
   public Future<SharedBudget> createBudget(SharedBudget sharedBudget, RequestContext requestContext) {
+    log.debug("createBudget:: Creating budget for shared budget id: {}", sharedBudget.getId());
     return fundFiscalYearService.retrievePlannedFiscalYear(BudgetUtils.convertToBudget(sharedBudget)
-      .getFundId(), requestContext)
+        .getFundId(), requestContext)
       .compose(plannedFiscalYear -> {
         Budget budget = BudgetUtils.convertToBudget(sharedBudget);
         if (plannedFiscalYear != null && budget.getFiscalYearId()
           .equals(plannedFiscalYear.getId())) {
+          log.info("createBudget:: Creating planned budget because planned fiscal year is not null and it is equal to budget fiscal year id");
           return createPlannedBudget(sharedBudget, requestContext);
         } else {
+          log.info("createBudget:: Creating new budget because planned fiscal year is null or it is not equal to budget fiscal year id");
           return createNewBudget(sharedBudget, requestContext);
         }
       })
-      .recover(this::processBudgetException);
+      .recover(e -> {
+        log.error("Failed to create budget", e);
```

Comment:
the same here, please review places by yourself that write errors or warns and add some identifier

Reply:
Thanks, added